File: src/api/aiDeliveryLog.ts
import request from './request'

/**
 * V0 §八 V0-6：AI 通知外部投递日志（dingtalk/email/wecom）。
 * SENT=渠道已受理；FAILED=渠道或网络失败；UNKNOWN=异步提交未确认。
 * 渠道本身不提供更进一步的送达/已读回执。
 */
export interface DeliveryLogEntry {
  id: number
  toolKey: string
  channel: 'dingtalk' | 'email' | 'wecom' | string
  caller: string | null
  conversationPk: number | null
  conversationId: string | null
  /** 收件目标脱敏摘要 */
  recipientSummary: string | null
  /** 标题/内容前 100 字预览 */
  subjectPreview: string | null
  status: 'SENT' | 'FAILED' | 'UNKNOWN' | string
  externalErrcode: string | null
  errorMessage: string | null
  attempts: number
  createdAt: string
  updatedAt: string
}

export interface DeliveryLogPage {
  records: DeliveryLogEntry[]
  total: number
}

export interface DeliveryLogQuery {
  page?: number
  size?: number
  toolKey?: string
  status?: string
  caller?: string
}

/** 分页查询 AI 通知投递日志（管理端） */
export function fetchDeliveryLogs(params: DeliveryLogQuery): Promise<DeliveryLogPage> {
  return request.get('/agent/delivery-log', { params })
}

File: backend/src/main/java/com/mftb/admin/entity/AiDeliveryLog.java
package com.mftb.admin.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * AI 通知外部投递日志（V0 §八 V0-6）。
 * <p>与 {@code ai_tool_exec_log} 区分：exec_log 关注「工具是否被允许执行」，
 * 本表关注「渠道是否已受理该次投递」。渠道不保证送达/已读，SENT 仅代表钉钉/SMTP 侧接收。
 */
@Data
@TableName("ai_delivery_log")
public class AiDeliveryLog {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String toolKey;
    /** dingtalk / email / wecom / ... */
    private String channel;
    private String caller;
    private Long conversationPk;
    private String conversationId;
    /** 收件目标脱敏摘要（如 a***@example.com，或钉钉群名） */
    private String recipientSummary;
    /** 标题或内容前 100 字预览 */
    private String subjectPreview;
    /** SENT / FAILED / UNKNOWN */
    private String status;
    /** 渠道返回码：钉钉 errcode 转字符串；SMTP 响应码等 */
    private String externalErrcode;
    private String errorMessage;
    private Integer attempts;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
